### ISSUE TITLE
0.0.3 gem doesn't work on Ruby 1.9.2, though fixed on master

### ISSUE THREAD

**Issue by mmcgrana** [headius/ruby-atomic#2]
The 0.0.3 gem doesn't work on 1.9.2 due to the weird behaviour of `defined?`:

On the commit that introduced 0.0.3:

```
$ git checkout -b test 186fb3cecfbeae4bb9f7cd3b218770a3e5443ab1
Switched to a new branch 'test'
$ rvm use 1.9.2
Using /Users/mmcgrana/.rvm/gems/ruby-1.9.2-p136
$ rake test
(in /Users/mmcgrana/state/repos/ruby-atomic)
<internal:lib/rubygems/custom_require>:33:in `require': no such file to load -- atomic_reference (LoadError)
    from <internal:lib/rubygems/custom_require>:33:in `rescue in require'
    from <internal:lib/rubygems/custom_require>:29:in `require'
    from /Users/mmcgrana/state/repos/ruby-atomic/lib/atomic.rb:46:in `<top (required)>'
    from <internal:lib/rubygems/custom_require>:29:in `require'
    from <internal:lib/rubygems/custom_require>:29:in `require'
    from test/test_atomic.rb:2:in `<top (required)>'
    from /Users/mmcgrana/.rvm/gems/ruby-1.9.2-p136/gems/rake-0.8.7/lib/rake/rake_test_loader.rb:5:in `load'
    from /Users/mmcgrana/.rvm/gems/ruby-1.9.2-p136/gems/rake-0.8.7/lib/rake/rake_test_loader.rb:5:in `block in <main>'
    from /Users/mmcgrana/.rvm/gems/ruby-1.9.2-p136/gems/rake-0.8.7/lib/rake/rake_test_loader.rb:5:in `each'
    from /Users/mmcgrana/.rvm/gems/ruby-1.9.2-p136/gems/rake-0.8.7/lib/rake/rake_test_loader.rb:5:in `<main>'
rake aborted!
Command failed with status (1): [/Users/mmcgrana/.rvm/rubies/ruby-1.9.2-p13...]

(See full trace by running task with --trace)
```

On master:

```
$ git co master
Switched to branch 'master'
~/state/repos/ruby-atomic(master) $ rake test
(in /Users/mmcgrana/state/repos/ruby-atomic)
Loaded suite /Users/mmcgrana/.rvm/gems/ruby-1.9.2-p136/gems/rake-0.8.7/lib/rake/rake_test_loader
Started
.......
Finished in 0.001251 seconds.

7 tests, 11 assertions, 0 failures, 0 errors, 0 skips

Test run options: --seed 39679
```

This is due to the Ruby/Jruby branching logic in that older commit:

```
$ rvm use 1.9.2
$ irb
ruby-1.9.2-p136 :001 > defined? RUBY_ENGINE && RUBY_ENGINE == "jruby"
 => "expression" 
$ rvm use 1.8.7
$ irb
ruby-1.8.7-p330 :002 > defined? RUBY_ENGINE && RUBY_ENGINE == "jruby"
 => nil
```

What do you think about cutting 0.0.4 with the most recent code?

**Comment by headius** [comment]
Missed this bug at some point. Will push an updated 0.0.4.

**Comment by headius** [comment]
Bump to 0.0.4 for release. Closed by 93975d8091205fecba74fd3f511114e74911ed71.

**Comment by kim** [comment]
Could you clarify which incarnation/version of MRI is needed for the C implementation of atomic reference? Tried different patchlevels of 1.9.1 and 1.9.2, but it always falls back to the ruby impl...
